### ISSUE TITLE
[Vue] When loading the page it loads all the Ionic icons even if not used

### ISSUE THREAD

**Issue by sneko** [ionic-team/ionic-framework#17886]
Hi,

I'm using the core page 4.1.1 and vue package 0.0.2.

I just succeeded to use Ionic with Vue.js instead of using `script/style` tags but I notice that from the first load of the page, my browser tries to download ALL the Ionic icons even if they are not used at all (dozens of .svg).

When developing locally that's fine but when reaching my remote server it takes a while due to too many connections.

I'm not sure that's on the Ionic Core side but maybe more about the Vue part. Does anyone is experiencing that with React/Angular?

Thank you,

Note: I would like a behavior like lazy loading for components, just load what you need on the fly

**Comment by sneko** [comment]
@mhartington do you think that's something I can fix from my side? Maybe not important global Ionic CSS style files?

**Comment by guillemmateos** [comment]
I'm having sort of the same issue, or at least very similar. I'm building a Vue app, and i'm using `@ionic/vue` and just by including ionic in the app, when I build the vue app, the dist folder has all the svg's for the full set of ionicons which is not desirable by any means.

If you then build your app as a PWA, the cache from the serviceworker will try and cache all those files, so you get to cache about 1k files, which is completely crazy.

@mhartington any suggestion on potential ways to address this? Is this maybe related to the includes in `ionic-vue.common.js`in lines 7 and 8? Would being specific on the `var icons = require('ionicons/icons')`import help fix this by importing only the relevant icons?

**Comment by obrunopolo** [comment]
I'm facing the same thing, but not using Vue. I'm working on a project based on Ionic PWA Toolkit with Stencil and observe the same: when building, it includes all ion-icons and then serviceworker precache all images in /build/svg. It would be desirable that we could choose which ionicons will be bundled. Not sure if this is an Ionic or Stencil issue, but since Stencil is from Ionic team I'm posting here

**Comment by sanzwebdevelopment** [comment]
Noticed the same with my Ionic/Angular PWA ... so the simple workaround I used is to add the following to assetGroups[] in ngsw-config.json:

`{
			"name": "icons",
			"installMode": "prefetch",
			"updateMode": "prefetch",
			"resources": {
				"files": [
					"!/svg/*",
					"/svg/home-outline.svg",
					"/svg/search-outline.svg",
					"/svg/people-outline.svg",
					"/svg/stats-chart-outline.svg",
					"/svg/information-circle-outline.svg",
					"/svg/help-circle-outline.svg",
					"/svg/warning-outline.svg",
					"/svg/heart-outline.svg",
					"/svg/arrow-forward.svg",
					"/svg/arrow-back.svg",
					"/svg/globe-outline.svg",
					"/svg/map-outline.svg",
					"/svg/bulb-outline.svg",
					"/svg/book-outline.svg",
					"/svg/close-outline.svg",
					"/svg/chevron-back.svg",
					"/svg/menu-outline.svg"
				]
			}
		}`

Basically block all icons from being prefetched, then specify only the ones included in the app.

See how that works for you :-)

**Comment by liamdebeasi** [comment]
Thanks for the issue! We recently released an all new Ionic Vue beta that should fix most of the issues reported. Can you try out the latest Ionic Vue beta and let me know if this is still an issue?



**Getting Started with the latest Ionic Vue beta**

The new Ionic Vue beta is built for Vue 3. Moving forward, we are not going to support Vue 2. For information on how to get started with the latest beta, check out our [Ionic Vue Getting Started Guide](https://ionicframework.com/docs/vue/quickstart).

At this time, we do not have a migration path from Ionic Vue built for Vue 2 to Ionic Vue built for Vue 3, but the Ionic Framework differences should be minimal.

Feel free to check out our [Ionic Vue Beta Announcement](https://ionicframework.com/blog/announcing-the-new-ionic-vue-beta/) blog post for more details on this release.



**Help! The reported issue still happens in the latest Ionic Vue beta.**

Please follow up here with the issue reproduced in an Ionic starter application running the latest Ionic Vue beta. Also include any revised steps to reproduce the issue. Issues that do not have a GitHub repo with the issue reproduced in an Ionic starter app will be closed.



**My issue is resolved/My issue is no longer relevant**.

You can either close the issue yourself, or do nothing. If there is no response within 14 days, this thread will be automatically closed.



**I have a new issue**.

Please [create a new issue](https://github.com/ionic-team/ionic-framework/issues/new/choose). Be sure to reproduce the issue in an Ionic starter application running the latest Ionic Vue beta and include a link to the repo.

Be sure to search for your issue first, as the issue you are running into may have already been reported.



We are excited to get closer to the Ionic Vue release and look forward to welcoming new Vue developers to the Ionic community. Please let me know if there are any questions. Thanks!
